cms/src/main/java/uk/nhs/digital/apispecs/ApiSpecificationImportMetadata.java

```diff
@@ -0,0 +1,140 @@
+package uk.nhs.digital.apispecs;
+
+import static java.util.Collections.unmodifiableCollection;
+
+import org.apache.commons.lang3.builder.*;
+
+import java.time.Instant;
+import java.util.*;
+
+/**
+ * Meta data of API Specifications processed during import.
+ */
+public class ApiSpecificationImportMetadata {
+
+    final Map<String, Item> apiSpecsMetadata = new HashMap<>();
+
+    private ApiSpecificationImportMetadata(final List<Item> apiSpecsMetadata) {
+
+        apiSpecsMetadata.forEach(metadataItem ->
+            this.apiSpecsMetadata.put(metadataItem.apiSpecJcrId(), metadataItem));
+    }
+
+    public static ApiSpecificationImportMetadata metadataWith(final List<Item> apiSpecsMetadata) {
+        return new ApiSpecificationImportMetadata(apiSpecsMetadata);
+    }
+
+    public Item getBySpecJcrId(final String specJcrId) {
+        if (!apiSpecsMetadata.containsKey(specJcrId)) {
+            apiSpecsMetadata.put(specJcrId, Item.item(specJcrId, Instant.EPOCH));
+        }
+
+        return apiSpecsMetadata.get(specJcrId);
+    }
+
+    public Collection<Item> items() {
+        return unmodifiableCollection(apiSpecsMetadata.values());
+    }
+
+    @Override public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        final ApiSpecificationImportMetadata that = (ApiSpecificationImportMetadata) o;
+        return apiSpecsMetadata.equals(that.apiSpecsMetadata);
+    }
+
+    @Override public int hashCode() {
+        return Objects.hash(apiSpecsMetadata);
+    }
+
+    @Override public String toString() {
+        return new ReflectionToStringBuilder(this, new MultilineRecursiveToStringStyle() {
+            {
+                setUseShortClassName(true);
+                setUseIdentityHashCode(false);
+            }
+        }).build();
+    }
+
+    /**
+     * Metadata of a single API Specification processed during import.
+     */
+    public static class Item {
+
+        private Instant lastChangeCheckInstant;
+        private final String apiSpecJcrId;
+        private boolean specExists;
+
+        private Item(final String apiSpecJcrId, final Instant lastChangeCheckInstant) {
+            this.lastChangeCheckInstant = lastChangeCheckInstant;
+            this.apiSpecJcrId = apiSpecJcrId;
+        }
+
+        public static Item item(final String apiSpecJcrId, final Instant lastChangeCheckInstant) {
```

Comment:
not sure what benefit this brings over just making the above constructor public?

Reply:
Admittedly, not much in this particular context, although this does help us avoid the noise of using keyword `new` everywhere.

It's just my personal preference, but it feels like this piece of test code reads _sligthly_ better (especially if we were to create a number of items in one go):
```
        final ApiSpecificationImportMetadata newApiSpecificationImportMetadata = metadataWith(
            item(apiSpecJcrId,  Instant.parse(oldLocalCheckTime))
        );
```

In general, however, I feel that factory methods tend to offer more expressiveness and control over object creation in comparison to always using constructor, and that's what has gotten me into habit of using them almost everywhere.